### PR TITLE
feat(datepicker): Rename props and update onChange signature

### LINKED ISSUE #30: onDateChange signature inconsistent with Semantic UI React components
# 🐛 bug report

## Description of the problem
This component has a very different API/parameters/signature for onDateChange event to that presented by the Semantic UI React components. The components in Semantic UI React all have `(event, data)` signatures for event handlers such as those for `onChange` . This facilitates consistent treatment for the whole family of components using generic event handlers.

## How has this issue affected you? What are you trying to accomplish?
The component simply doesn't play nicely in a form where generic event handlers have been written to interact with the events of Semantic UI React components.

### PR DESCRIPTION
**Pull request by arthurdenner**
**What kind of change does this PR introduce?**
Feature. Closes #30.

**What is the current behavior?**
Our component has a different API from the other components from React Semantic UI.
The main differences are:

- Components from RSUI have a prop named `onChange`;
- Components from RSUI have a prop named `value`;
- The signature for the `onChange` props on RSUI is: (event: SyntheticEvent, data: ComponentProps).

**What is the new behavior?**
All the points mentioned above are addressed.

**Checklist**:

- [x] Documentation
- [x] Tests
- [ ] Ready to be merged

### CODE REVIEW

**Comment by arthurdenner** [comment]
> Why is the event not passed as the first parameter in `basic` and `range`?

No reason. I just happened to add it as the second argument.

Do you think it's better to have it as the first for consistency or any other reason?

**Comment by emersonlaurentino** [comment]
> Do you think it's better to have it as the first for consistency or any other reason?

Yes, once React Semantic UI uses this pattern.
`f(event: SyntheticEvent, data: object)`

File: src/types/index.ts [hunk 1]
```diff
@@ -76,18 +75,18 @@ export type DayzedProps = {
   minDate?: Date;
   monthsToDisplay: number;
   offset: number;
-  onDateSelected: (props: any) => void;
+  onDateSelected: (dateObj: any, event: React.SyntheticEvent) => void;
```
Comment (emersonlaurentino) [inline src/types/index.ts]: shouldn't this be the event as the first parameter?

Comment (arthurdenner) [reply src/types/index.ts]: no, this is the type for the `dayzed` library, where the event is the second parameter...

**Pull request merged**